src/go/rpk/pkg/cli/cloud/mcp/mcp.go

```diff
@@ -0,0 +1,210 @@
+				expired, err := oauth.ValidateToken(authToken, acl.Audience(), authVir.ClientID)
+				if err != nil {
+					m.Lock()
+					tokenOK = false
+					m.Unlock()
+					return fmt.Errorf("failed to validate Cloud token. This may happen, if they never logged in to Redpand Cloud. However, their token is not just expired, we'd not see this error then. Instruct the user to run `rpk%s cloud login` to login to Redpanda Cloud. After running the command, they can ask to retry.", extra)
+				}
+				if expired {
+					m.Lock()
+					tokenOK = false
+					m.Unlock()
+					return fmt.Errorf("Redpanda Cloud token is expired. Instruct the user to run `rpk%s cloud login` to obtain a fresh one. Afterwards, they can ask to retry.", extra) //nolint:revive // We want it to start with capital letter, because Redpanda is a name
+				}
+				m.RLock()
+				ok := tokenOK
+				m.RUnlock()
+				if !ok {
+					cl.UpdateAuthToken(authToken)
+					m.Lock()
+					tokenOK = true
+					m.Unlock()
+					fmt.Fprintf(os.Stderr, "Loaded new token\n")
+				}
+				return nil
+			}
+
+			s := server.NewMCPServer(
+				"Redpanda Cloud MCP",
+				"beta",
+				server.WithToolHandlerMiddleware(func(thf server.ToolHandlerFunc) server.ToolHandlerFunc {
+					return func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+						if !allowDelete && strings.Contains(strings.ToLower(request.Params.Name), "delete") {
+							return nil, fmt.Errorf("deletes are forbidden. Tell the user that they can enable deletes by running rpk cloud mcp install --allow-delete. This will permanently turn on deletes via MCP, which is off by default. Then, they must restart their MCP client. It is important that you tell them to restart their client. To turn it back off, they can run it without the --allow-delete flag.")
+						}
+						if err := maybeReloadToken(); err != nil {
+							return nil, err
+						}
+
+						return thf(ctx, request)
+					}
+				}),
+			)
+
+			// Controlplane
+			controlplanev1mcp.ForwardToConnectRegionServiceClient(s, cl.Region)
+			controlplanev1mcp.ForwardToConnectResourceGroupServiceClient(s, cl.ResourceGroup)
+			controlplanev1mcp.ForwardToConnectClusterServiceClient(s, cl.Cluster)
+			controlplanev1mcp.ForwardToConnectNetworkServiceClient(s, cl.Network)
+			controlplanev1mcp.ForwardToConnectServerlessClusterServiceClient(s, cl.Serverless)
+			controlplanev1mcp.ForwardToConnectServerlessRegionServiceClient(s, cl.ServerlessRegion)
+			controlplanev1mcp.ForwardToConnectOperationServiceClient(s, cl.Operation)
+			controlplanev1mcp.ForwardToConnectServerlessRegionServiceClient(s, cl.ServerlessRegion)
+			controlplanev1mcp.ForwardToConnectServerlessRegionServiceClient(s, cl.ServerlessRegion)
+
+			if err := server.ServeStdio(s); err != nil {
+				fmt.Fprintf(os.Stderr, "Server error: %v\n", err)
+			}
+		},
+	}
+
+	cmd.Flags().BoolVarP(&allowDelete, "allow-delete", "", false, "Allow delete RPCs. Off by default.")
```

Comment:
```suggestion
	cmd.Flags().BoolVarP(&allowDelete, "allow-delete", "", false, "Allow delete RPCs. Off by default")
```

We don't add a period in the flag help text (sorry, we don't have a linter for this yet).

Reply:
done